### PR TITLE
Fix combining module with folder/project creation

### PR DESCRIPTION
**Pull request by averbuks**
Fix combining module with folder/project creation #19 by explicitly specifying resource hierarchy node type with `policy_for` variable. 
`Clean up integration tests` #20 already merged into the PR code

### CODE REVIEW

File: docs/upgrading_to_v3.0.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Upgrading to v3.0
+
+The v3.0 release of *org-policy* is a backwards incompatible
+release.
+
+## Migration Instructions
+
+In previous versions of this module the resource hirerarchy node was not defined explicitly which caused problems when the module is combined with project/folder creation. The `policy_for` variable was introdused to address that issue.
```
Comment (ludoo) [inline docs/upgrading_to_v3.0.md]: Typo: `s/introdused/introduced/`

Comment (averbuks) [reply docs/upgrading_to_v3.0.md]: @ludoo thank you for being so attentive, that's fixed

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,8 +24,8 @@ Please add new unreleased features here.
 
 This is the initial release of the module, with full support for org policy management.
 
-[Unreleased]: https://github.com/terraform-google-modules/terraform-google-org-policy/compare/v2.0.1...HEAD
-[2.0.0]: https://github.com/terraform-google-modules/terraform-google-org-policy/compare/v2.0.0...v2.0.1
+[Unreleased]: https://github.com/terraform-google-modules/terraform-google-org-policy/compare/v3.0.0...HEAD
+[3.0.0]: https://github.com/teHEADrraform-google-modules/terraform-google-org-policy/compare/v2.0.0...v3.0.0
```
Comment (aaron-lane) [inline CHANGELOG.md]: This looks like an issue was introduced during a rebase.

```suggestion
[3.0.0]: https://github.com/terraform-google-modules/terraform-google-org-policy/compare/v2.0.0...v3.0.0
```

Comment (averbuks) [reply CHANGELOG.md]: Good catch! Fixed

**Pull request merged**